addons/website_slides/wizard/base_partner_merge.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+
+from odoo import api, models
+
+
+class MergePartnerAutomatic(models.TransientModel):
+    _inherit = 'base.partner.merge.automatic.wizard'
+
+    @api.model
+    def _merge_partner_with_preferences(self, src_partners, dst_partner,
+                                        res_model, uniq_field, preferences, keep_dst_partner=False):
+        """ Merge records of "table" for the given partners following the given preferences.
+
+        ! It is the caller responsibility to check the ACL, this method is executed in sudo.
+        Use this method when the table contains a unique key per partner to select the
+        records to assign to the dst_partner from the record of src_partners and dst_partner.
+        Indeed, when there is an uniq constraint on partner_id and another column we cannot
+        simply assign all record of src_partners to dst_partner without potentially create
+        conflicts.
+        When multiple records with the same unique key per partner among the record of the
+        given partners (src_partners and dst_partner) are encountered, this method choose
+        the one with the highest preference and assign it to dst_partner. Of course, if there
+        is only one, it is just assigned to the dst_partner.
+
+        :param recordset src_partners: source partners
+        :param recordset dst_partner: destination partner
+        :param str res_model: model name
+        :param str uniq_field: column name of the unique key per partner
+        :param str preferences: order by clause (ex.: completed DESC, vote DESC)
+        :param bool keep_dst_partner: when a conflict occurs whether to keep the record with
+        dst_partner if present instead of following the preferences
+        """
+        Model = self.env[res_model].sudo()
+        record_ids_to_delete = list()
+        prev_uniq_id = None
+        current_id_kept = None
+        for record in Model.search_read([('partner_id', 'in', (src_partners + dst_partner).ids)],
+                                        [uniq_field, 'partner_id'],
+                                        order=uniq_field + ((',' + preferences) if preferences else '')):
```

Comment:
Why not set `preferences` to `''` by default instead of requiring to call with `''` as it is a real use case we have below?

(also, fewer parentheses with
```suggestion
                                        order=uniq_field + (f', {preferences}' if preferences else ''):
```
)